core/src/apps/ethereum/yielding.py

```
@@ -209,81 +225,82 @@ def _prepare_claim_rewards(
-            token_labels.append(label)
-
-    except (ValueError, EOFError, InvalidFunctionCall):
+        if len(users) == 0:
+            raise ValueError
+
+        first_user = users[0]
+        if not isinstance(first_user, bytes):
+            raise ValueError
+
+    except (ValueError, InvalidFunctionCall):
         raise DataError("Invalid data for claim rewards transaction")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)